dangerzone/gui/main_window.py

```diff
@@ -694,24 +662,33 @@ def closeEvent(self, e: QtGui.QCloseEvent) -> None:
 class WaitingWidget(QtWidgets.QWidget):
     def __init__(self) -> None:
         super().__init__()
-        self.label = QtWidgets.QLabel()
-        self.label.setAlignment(QtCore.Qt.AlignCenter)
-        self.label.setTextFormat(QtCore.Qt.RichText)
-        self.label.setOpenExternalLinks(True)
-        self.label.setWordWrap(True)
+        self.header = QtWidgets.QLabel()
+        self.header.setAlignment(QtCore.Qt.AlignCenter)
+        self.footer = QtWidgets.QLabel()
+        self.footer.setAlignment(QtCore.Qt.AlignRight)
+        self.footer.setTextFormat(QtCore.Qt.RichText)
+        self.footer.setOpenExternalLinks(True)
+        self.footer.setWordWrap(True)
+        self.traceback_widget = TracebackWidget()
+        self.traceback_widget.setVisible(False)
         layout = QtWidgets.QVBoxLayout()
-        layout.addWidget(self.label)
+        layout.addWidget(self.header)
+        layout.addWidget(self.traceback_widget)
+        layout.addWidget(self.footer)
+        layout.addStretch()
         self.setLayout(layout)
 
-    def handle_start(self) -> None:
-        # FIXME: The following message is a placeholder, we need to find a more
-        # descriptive one.
-        self.label.setText(
-            "Oh hi there!<br><br>"
-            "First time, huh?<br><br>"
-            "Welcome! I'm afraid you gonna have to wait a bit.<br>"
-            "Check the bottom-right corner for a progress report"
-        )
+    def handle_error(self, msg: str) -> None:
+        self.header.setText("<p>Dangerzone encountered an error during startup:<p>")
+        self.header.setStyleSheet("QLabel { font-size: 20px; }")
+        self.traceback_widget.process_output(msg)
+        self.footer.setText("""\
+            <p>
+              <a href="https://github.com/freedomofpress/dangerzone/wiki/Reporting-an-issue">
+                 Report issue ↗️
```

Comment:
↗️ should not be inside the link. And I believe we could just get rid of it entirely, or use it consistently for all our links.

Reply:
I've thought about this as well. We currently use buttons for actions in the UI, but here we redirect the user to another site. So it's more of a link to me, but if you feel it's confusing, I can change it.